components/scanners/snyk/component.yaml

```diff
@@ -0,0 +1,26 @@
+name: snyk
+description: "Runs snyk open source then parses findings into the OCSF format"
+type: scanner
+parameters:
+  - name: snyk_token
+    type: string
+    value: ""
+steps:
+  - name: run-snyk
+    image: components/scanners/snyk/scanner
+    env_vars:
+      SNYK_INTEGRATION_VERSION: docker
+      SNYK_INTEGRATION_NAME: smithy
+      SNYK_TOKEN: "{{.parameters.snyk_token}}"
+    executable: /bin/snyk
+    args:
+      - test
+      - --prune-repeated-subdependencies 
+      - --skip-unresolved
+      - --all-projects
+      - --sarif-file-output={{scratchWorkspace}}/snyk.out
+      - "{{sourceCodeWorkspace}}/"
```

Comment:
The CLI seems to exit with status code `1` if it finds vulnerabilities: https://docs.snyk.io/snyk-cli/commands/test#exit-codes

Which then causes Smithy to assume that the scan failed:

```
Error: unexpected failure: unexpected workflow execution failure: failed to run step for component 'snyk': failed to execute step 'run-snyk': failed to wait for container: container exited with an error status code: 1
```

Reply:
Perhaps a separate `entrypoint.sh` script that deals with non-`0` exit codes can resolve this.